app/services/promotions/keto_service.rb

```diff
@@ -13,10 +17,15 @@ def get_price(params)
         elsif not_keto_count >= 5
           not_keto_product_prices.sum * 0.25
         end
+        # CON: Usage of 3 different maps, could solve in a single one:
+        #
+        # total_sum = params['lineItems'].sum { |item| item[:price].to_f }
+        # final_price = (total_sum - descount).round(2)
+        #
```

Comment:
yes, change 
(params['lineItems'].pluck(:price).map{|n|n.to_f}.sum 
into =>
params['lineItems'].sum{|n|n[:price].to_f} 
and case when 
would be so much better, thanks for sugestion